services/delegation/src/modules/blockchain/blockchain.service.ts

```diff
@@ -110,6 +124,43 @@ export class BlockchainService {
     return false;
   }
 
+  async undelegate(address: string, amount: ethers.BigNumber) {
+    try {
+      const tx = await this.contract.undelegate(address, {
+        gasPrice: this.provider.getGasPrice(),
+        value: amount,
+      });
+      await tx.wait();
+      if (tx) {
+        const receipt = await this.provider.getTransactionReceipt(tx.hash);
```

Comment:
I think if you save the output of `await tx.wait();`, that should be the receipt. No need to call this again

Reply:
Yep.